src/Propel/Generator/Model/Table.php

```diff
@@ -791,7 +791,7 @@ public function setupReferrers($throwErrors = false)
                 $foreignPrimaryKeys = $foreignTable->getPrimaryKey();
                 // check all keys are referenced in foreign key
                 foreach ($foreignPrimaryKeys as $foreignPrimaryKey) {
-                    if (!$foreignPrimaryKey->hasReferrer($foreignKey) && $throwErrors) {
+                    if (!$foreignPrimaryKey->hasReferrer($foreignKey) && !$foreignKey->isSkipSql() && $throwErrors) {
```

Comment:
well, generating SQL has nothing to do with checking for valid foreign keys. So this check is a misuse of the `skip-sql` attribute to allow models generation with invalid foreign keys.